File: practice/problem-solving/arrays-hashing/contains-duplicate.py
class Solution:
    # 1
    # Hash Set
    # Time Complexity: O(n)
    # Space Complexity: O(n)
    def hasDuplicate (self, nums: List[int]) -> int:
        seen = set()
        for num in nums:
            if (num in seen):
                return True
            seen.add(num)
        return False

File: practice/problem-solving/arrays-hashing/contains-duplicate.js
class Solution {
    // 1
    // Hash Set
    // Time Complexity: O(n)
    // Space Complexity: O(n)
    hasDuplicate(nums) {
        const seen = new Set();
        for (const num of nums) {
            if (seen.has(num)) return true;
            seen.add(num);
        }
        return false;
    }
}
